src/plugins/modifier/modifier.ts

```diff
@@ -0,0 +1,80 @@
+import { Session } from "@azure-tools/autorest-extension-base";
+import { CodeModel } from "@azure-tools/codemodel";
+import { CliDirectiveManager } from "./cliDirective";
+
+export class Modifier {
+    private manager: CliDirectiveManager;
+
+    get codeModel() {
+        return this.session.model;
+    }
+
+    constructor(protected session: Session<CodeModel>) {
+    }
+
+    async init(): Promise<Modifier> {
+        this.manager = new CliDirectiveManager();
+        await this.manager.LoadDirective(this.session);
+        return this;
+    }
+
+    public process(): CodeModel {
+
+        let choices = [this.codeModel.schemas.choices ?? [], this.codeModel.schemas.sealedChoices ?? []];
+        choices.forEach(arr => {
+            arr.forEach(s => {
+                this.manager.process({
+                    choiceSchemaName: s.language.default.name,
+                    target: s,
+                });
+                s.choices?.forEach(ss => {
+                    this.manager.process({
+                        choiceSchemaName: s.language.default.name,
+                        choiceValueName: ss.language.default.name,
+                        target: ss,
+                    });
+                })
+            })
+        });
+
+        this.codeModel.schemas.objects.forEach(s => {
+            this.manager.process({
+                objectSchemaName: s.language.default.name,
+                target: s,
+            });
+            s.properties?.forEach(p => {
+                this.manager.process({
+                    objectSchemaName: s.language.default.name,
+                    propertyName: p.language.default.name,
+                    target: p,
+                })
+            });
+        });
+
```

Comment:
Do we need to deal with the child properties or parent properties for object schema?

Reply:
And how about globalParameter ？

Reply:
i think these properties can be changed by their schema object directly, so recursive support seems not needed here. feel free to let me know if it's not true
globalParameter is not support, please file an issue to me if you need it. I can support it in next PR. thx